pdfbox/src/test/java/org/apache/pdfbox/pdmodel/font/TestFontEmbedding.java

```diff
@@ -371,4 +376,116 @@ void testReuseEmbeddedSubsettedFont() throws IOException
             assertEquals(text1 + " " + text2, extractedText.trim());
         }
     }
+
+    private class TrueTypeEmbedderTester extends TrueTypeEmbedder {
+        /**
+         * Common functionality for testing the TrueTypeFontEmbedder
+         *
+         */
+        TrueTypeEmbedderTester(PDDocument document, COSDictionary dict, TrueTypeFont ttf, boolean embedSubset) throws IOException {
+            super(document, dict, ttf, embedSubset);
+        }
+
+        @Override
+        protected void buildSubset(InputStream ttfSubset, String tag, Map<Integer, Integer> gidToCid) throws IOException {
+            // no-op.  Need to define method to extend abstract class, but
+            // this method is not currently needed for testing
+        }
+    }
+
+    /**
+     * Test that we validate embedding permissions properly for all legal permissions combinations
+     *
+     * @throws IOException
+     */
+    @Test
+    void testIsEmbeddingPermittedMultipleVersons() throws IOException
+    {
+        // SETUP
+        PDDocument doc = new PDDocument();
+        COSDictionary cosDictionary = new COSDictionary();
+        InputStream input = PDFont.class.getResourceAsStream("/org/apache/pdfbox/resources/ttf/LiberationSans-Regular.ttf");
+        TrueTypeFont ttf = new TTFParser().parseEmbedded(input);
+        TrueTypeEmbedderTester tester = new TrueTypeEmbedderTester(doc, cosDictionary, ttf, true);
+        TrueTypeFont mockTtf = Mockito.mock(TrueTypeFont.class);
```

Comment:
I wanted to test with an actual copy of the ArialRoundedMTBold.ttf file which is the one in which we first detected this permissions problem.  However, I haven't heard back from legal about whether I'm allowed to redistribute a copy of that ttf file.  So, for now, I'm doing as much as I can with mocking instead.

Reply:
Yeah, long term we don't want to embed a problem font in their repo.

Reply:
I just heard back from Rachael Abdelkhalek, and she got a response from legal.  We can't just send the ttf file to the pdfbox maintainers.